### PR TITLE
Update product endpoint and test cases.

### PR DESCRIPTION
**Pull request by tasawernawaz**
ECOM-2234
- Update product endpoint for nested stock records.
- Fix broken test cases.

Please review this @ahsan-ul-haq @aamir-khan @awais786 @zubair-arbi

### CODE REVIEW

**Comment by ahsan-ul-haq** [comment]
:+1:

**Comment by tasawernawaz** [comment]
@clintonb @rlucioni Please review this.

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -54,12 +55,20 @@ class Meta(object):
...
+        fields = ('id', 'partner', 'partner_sku', 'price_currency', 'price_excl_tax',)
+
+
 class ProductSerializer(serializers.HyperlinkedModelSerializer):
     """ Serializer for Products. """
     attribute_values = ProductAttributeValueSerializer(many=True, read_only=True)
     product_class = serializers.SerializerMethodField()
     price = serializers.SerializerMethodField()
     is_available_to_buy = serializers.SerializerMethodField()
+    stockrecords = StockRecordSerializer(many=True, read_only=True)
```
Comment (clintonb) [inline ecommerce/extensions/api/serializers.py]: stock_records

File: ecommerce/extensions/api/v2/tests/views/__init__.py [hunk 2]
```diff
@@ -69,3 +70,15 @@ def serialize_product(self, product):
         })
 
         return data
+
+    def get_stock_records(self, product):
```
Comment (clintonb) [inline ecommerce/extensions/api/v2/tests/views/__init__.py]: This should be `def serialize_stock_records(self, stock_records)`, and the calling method should pass in `product.stockrecords.all()`.

File: ecommerce/extensions/api/serializers.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 Order = get_model('order', 'Order')
 Product = get_model('catalogue', 'Product')
 Partner = get_model('partner', 'Partner')
+StockRecord = get_model('partner', 'StockRecord')
```
Comment (clintonb) [inline ecommerce/extensions/api/serializers.py]: Alphabetize. Move this down.

**Comment by tasawernawaz** [comment]
@clintonb I have updated code, Please review it.

File: ecommerce/extensions/api/serializers.py [hunk 4]
```diff
@@ -23,6 +23,7 @@
 Partner = get_model('partner', 'Partner')
 ProductAttributeValue = get_model('catalogue', 'ProductAttributeValue')
 Refund = get_model('refund', 'Refund')
+StockRecord = get_model('partner', 'StockRecord')
```
Comment (clintonb) [inline ecommerce/extensions/api/serializers.py]: This remains unalphabetized.

**Comment by clintonb** [comment]
:+1: (once model list alphabetized)

**Comment by tasawernawaz** [comment]
@clintonb  Do we need to alphabetized according to app name ? Because models are already alphabetized.

File: ecommerce/extensions/api/serializers.py (hunk 4 above)
Comment (clintonb) [inline ecommerce/extensions/api/serializers.py]: Selector should come before StockRecord.

Comment (tasawernawaz) [inline ecommerce/extensions/api/serializers.py]: Code updated.

**Comment by rlucioni** [comment]
@tasawernawaz please delete merged branches.

**Pull request merged**